spacy/cli/debug_data.py

```diff
@@ -670,6 +671,51 @@ def debug_data(
                 f"Found {gold_train_data['n_cycles']} projectivized train sentence(s) with cycles"
             )
 
+    if "trainable_lemmatizer" in factory_names:
+        msg.divider("Trainable Lemmatizer")
+        trees_train: Set[str] = gold_train_data["lemmatizer_trees"]
+        trees_dev: Set[str] = gold_dev_data["lemmatizer_trees"]
+        # This is necessary context when someone is attempting to interpret whether the
+        # number of trees exclusively in the dev set is meaningful.
+        msg.info(f"{len(trees_train)} lemmatizer trees generated from training data")
+        msg.info(f"{len(trees_dev)} lemmatizer trees generated from dev data")
+        dev_not_train = trees_dev - trees_train
+
+        if len(dev_not_train) != 0:
+            pct = len(dev_not_train) / len(trees_dev)
+            msg.info(
+                f"{len(dev_not_train)} lemmatizer trees ({pct*100:.1f}% of dev trees)"
+                " were found exclusively in the dev data."
+            )
+        else:
+            # Would we ever expect this case? It seems like it would be pretty rare,
+            # and we might actually want a warning?
+            msg.good("All trees in dev data present in training data.")
+
+        if gold_train_data["n_low_cardinality_lemmas"] > 0:
+            n = gold_train_data["n_low_cardinality_lemmas"]
+            msg.warn(f"{n} docs with 1 or 0 unique lemmas.")
+        else:
+            msg.good("All training docs meet lemma uniqueness requirements.")
+
+        if gold_dev_data["n_low_cardinality_lemmas"] > 0:
+            n = gold_dev_data["n_low_cardinality_lemmas"]
+            msg.warn(f"{n} docs with 1 or 0 unique lemmas.")
+        else:
+            msg.good("All dev docs meet lemma uniqueness requirements.")
+
+        if gold_train_data["no_lemma_annotations"] > 0:
+            n = gold_train_data["no_lemma_annotations"]
+            msg.warn(f"{n} docs with no lemma annotations.")
+        else:
+            msg.good("All training docs have complete lemma annotations.")
```

Comment:
```suggestion
            msg.good("All training docs have lemma annotations.")
```

_complete_ suggests that every token is annotated. We'd still allow partial annotations.

Reply:
Ah, the check I currently have is `all(token.lemma == 0 for token in gold)` for this, so I think the current message makes sense. But I'll also add some lines for identifying the number of docs with partial annotations. That would be the similar logic but switching `all` to `any`, then I think changing the message when `count > 0` from a `warn` to an `info`.

Reply:
Updated in latest commit.